classes/auth.php

```diff
@@ -594,14 +596,19 @@ public function saml_login_complete($attributes) {
 
         // Find Moodle user.
         $user = null;
-        foreach ($attributes[$attr] as $uid) {
-            if ($this->config->tolower) {
-                $this->log(__FUNCTION__ . " to lowercase for $uid");
-                $uid = strtolower($uid);
-            }
-            if ($user = get_complete_user_data($this->config->mdlattr, $uid)) {
-                // We found a user.
-                break;
+        if ($this->config->matchnameid) {
+            $uid = $nameid->value;
```

Comment:
Would be good to have test covering this case (the rest of `saml_login_complete` is test covered already).

Reply:
Removed this option as nameidasattribute covers it